parsec/backend/pki.py

```diff
@@ -214,117 +235,106 @@ async def api_pki_enrollment_info(
         try:
             await self.reject(
                 organization_id=client_ctx.organization_id,
-                enrollment_id=msg["enrollment_id"],
+                enrollment_id=msg.enrollment_id,
                 rejected_on=DateTime.now(),
             )
-            rep = {"status": "ok"}
+            return PkiEnrollmentRejectRepOk()
 
         except PkiEnrollmentNotFoundError as exc:
-            rep = {"status": "not_found", "reason": str(exc)}
+            return PkiEnrollmentRejectRepNotFound(str(exc))
 
         except PkiEnrollmentNoLongerAvailableError as exc:
-            rep = {"status": "no_longer_available", "reason": str(exc)}
-
-        return pki_enrollment_reject_serializer.rep_dump(rep)
+            return PkiEnrollmentRejectRepNoLongerAvailable(str(exc))
 
     @api("pki_enrollment_accept")
     @catch_protocol_errors
+    @api_typed_msg_adapter(PkiEnrollmentAcceptReq, PkiEnrollmentAcceptRep)
     async def api_pki_enrollment_accept(
-        self, client_ctx: AuthenticatedClientContext, msg: dict[str, object]
-    ) -> dict[str, object]:
+        self, client_ctx: AuthenticatedClientContext, msg: PkiEnrollmentAcceptReq
+    ) -> PkiEnrollmentAcceptRep:
         if client_ctx.profile != UserProfile.ADMIN:
-            return {
-                "status": "not_allowed",
-                "reason": f"User `{client_ctx.device_id.user_id.str}` is not admin",
-            }
-        msg = pki_enrollment_accept_serializer.req_load(msg)
+            return PkiEnrollmentAcceptRepNotAllowed(
+                f"User `{client_ctx.device_id.user_id.str}` is not admin",
+            )
 
         try:
-            PkiEnrollmentAnswerPayload.load(msg["accept_payload"])
+            PkiEnrollmentAnswerPayload.load(msg.accept_payload)
 
         except DataError as exc:
-            return {"status": "invalid_payload_data", "reason": str(exc)}
+            return PkiEnrollmentAcceptRepInvalidPayloadData(str(exc))
 
         try:
             user, first_device = validate_new_user_certificates(
                 expected_author=client_ctx.device_id,
                 author_verify_key=client_ctx.verify_key,
-                device_certificate=msg["device_certificate"],
-                user_certificate=msg["user_certificate"],
-                redacted_user_certificate=msg["redacted_user_certificate"],
-                redacted_device_certificate=msg["redacted_device_certificate"],
+                device_certificate=msg.device_certificate,
+                user_certificate=msg.user_certificate,
+                redacted_user_certificate=msg.redacted_user_certificate,
+                redacted_device_certificate=msg.redacted_device_certificate,
             )
 
         except CertificateValidationError as exc:
-            return pki_enrollment_accept_serializer.rep_dump(
-                {"status": exc.status, "reason": exc.reason}
-            )
+            return PkiEnrollmentAcceptRepInvalidCertification(exc.reason)
```

Comment:
Do we really want to keep the reason ?

Reply:
I followed the protocol described in `oxidation/libparsec/crates/protocol/schema/`

Reply:
I mean `reason` is still here, but we can return `None`

Reply:
> Do we really want to keep the reason ?

This is a migration PR, so let's keep the fields as they are. We want to keep the same behavior as before in order to avoid issues. The possible removing of the `reason` fields should be dealt with properly later, by updating the schemas.